http/src/util/service/handler.rs

```diff
@@ -144,37 +173,45 @@ from_req_impl! { Extract7; A, B, C, D, E, F, G }
 from_req_impl! { Extract8; A, B, C, D, E, F, G, H }
 from_req_impl! { Extract9; A, B, C, D, E, F, G, H, I }
 
-pub trait Handler<Arg>: Clone {
+/// Same as `std::ops::Fn` trait but for async output.
+///
+/// It is necessary in the the HRTB bounds for async fn's with reference paramters because it
+/// allows the output future to be bound to the paramter lifetime.
+///     `F: for<'a> AsyncFn<(&'a u8,) Output=u8>`
+pub trait AsyncFn<Arg> {
```

Comment:
Separated to be reused elsewhere when needed. I'm thinking of tranform_fn if you like @fakeshadow .

Reply:
It's fine. I don't mind move it to service crate or even duplicate it there.

Reply:
Ok. then I'll try the same thing for transform_fn to remove the cloneability constraint.